code/modules/mob/mob_movement.dm

```diff
@@ -194,8 +194,10 @@
 
 		if(mob.confused && IsEven(world.time))
 			step(mob, pick(cardinal))
+			mob.last_movement=world.time
 		else
 			. = ..()
+			mob.last_movement=world.time

```

Comment:
Why are you adding it to both sides of if when you can just place it once outside?
Not to say i thought that it is already set somewhere, i doubt this var was unused before